base/abstractarray.jl

```diff
@@ -3503,11 +3503,29 @@ julia> map!(+, zeros(Int, 5), 100:999, 1:3)
 ```
 """
 function map!(f::F, dest::AbstractArray, As::AbstractArray...) where {F}
-    isempty(As) && throw(ArgumentError(
-        """map! requires at least one "source" argument"""))
+    @assert !isempty(As) # should dispatch to map!(f, A)
     map_n!(f, dest, As)
 end
 
+"""
+    map!(function, array)
+
+Like [`map`](@ref), but stores the result in the same array.
+
```

Comment:
```suggestion

!!! compat "Julia 1.12"
    This method requires Julia 1.12 or later. To support previous versions, 
    use `map!(function, array, array)`

```

Reply:
Do we want to note the old behaviour? 
```
!!! compat "Julia 1.12"
    This method requires Julia 1.12 or later. To support previous versions
    use `map!(function, array, array)`. Note that on Julia 1.9 and earlier,
    `map!(f, array)` had a different meaning, equivalent to `map(_ -> f(), array, array)`.
    That method was removed from Julia 1.10.
```
But I'm not sure this is correct... actually I thought this was removed in 1.11, but see the error in 1.10:
```julia
julia> VERSION
v"1.10.7"

julia> map!(rand, zeros(1,3))
ERROR: ArgumentError: map! requires at least one "source" argument
```
Don't have 1.9 on this computer right now... and maybe we don't care about < LTS versions anyway?

Reply:
Tweak to make clearer that the compat suggestion works before and after:
```suggestion
!!! compat "Julia 1.12"
    This method requires Julia 1.12 or later. To support previous versions too,
    use the equivalent `map!(function, array, array)`.
```